### ISSUE TITLE
How can get last cves in a specify vendor?

### ISSUE THREAD

**Issue by zJvco** [cve-search/CveXplore#73]
Hello

I install this tool now, how can I get last cves by vendor?

**Comment by P-T-I** [comment]
Did you've taken a look at the [documentation?](https://cve-search.github.io/CveXplore/) Are you planning to use the CLI or the PACKAGE?

**Comment by zJvco** [comment]
Hello, I think want to use a package, my idea is to use this too to get the last cves by vendor. I read the documentation and I dont look nothing with this.

**Comment by P-T-I** [comment]
Ok; as the documentation states:

>>> from CveXplore import CveXplore
>>> cvx = CveXplore()
>>> result = cvx.cves.get_cves_for_vendor("microsoft", limit=10)

**Comment by zJvco** [comment]
Yes, I did that, but these cvs are from 2018, I want all cvs recent... I can do that?

**Comment by P-T-I** [comment]
That’s because they are sorted by cvss by default!

If you want the cves based on modified date, you could try for instance:

cvx.cves.vendor.find(“microsoft”).sort("Modified", pymongo.DESCENDING)

**Comment by zJvco** [comment]
This is gave me an error:

"AttributeError: 'CvesDatabaseFunctions' object has no attribute 'vendor' "

my code:

from CveXplore import CveXplore
cvx = CveXplore(mongodb_connection_details={"host": "localhost", "port": 27017})
cvx.cves.vendor.find("microsoft").sort("Published", pymongo.DESCENDING)

**Comment by P-T-I** [comment]
Sorry; doing this from the top of my head; I made a typo; vendor should be vendors:

cvx.cves.vendors.find("microsoft").sort("Published", pymongo.DESCENDING)

**Comment by zJvco** [comment]
Hi man.

this works, but it doesn't bring all the vendors cves. I need to find this cves CVE-2021-38647, CVE-2021-38648, CVE-2021-38645, CVE-2021-38649.

The filtering method is by vendors, and from what I've seen, in the database, some cve don't have this field... How can this be fixed?

**Comment by P-T-I** [comment]
@zJvco Simple; it cannot.... At least not easily; the reason that there is no vendor assigned is because there are no CPE's assigned to this particular CVE's. The updater of CveXplore determines the vendor based on the assigned cpe; and the CVE's content is determined by nist.gov (which is the source of the CveXplore database). If you [check](https://nvd.nist.gov/vuln/detail/CVE-2021-38645) you will see that this particular cve is still undergoing analysis; once that is completed I suspect that the Known Affected Software Configurations of the CVE is filled; like in this [particular](https://nvd.nist.gov/vuln/detail/CVE-2021-1810) CVE. At a given interval nist.gov releases new updates and the CveXplore updater will get those updates and insert them into the database.

So to recap; be patient.... It will be in there....

**Comment by zJvco** [comment]
Hmm, thank, I really did not know that... Ok I wait for that

**Comment by P-T-I** [comment]
@zJvco I'm sorry; I kinda mist your last question... Sure that will work!
